packages/cw-orch-mock/src/core.rs

```diff
@@ -71,113 +70,268 @@ pub type MockApp = App<
+                })
+        }
+
+        /// Query the (bank) balance of a native token for and address.
+        /// Returns the amount of the native token.
+        pub fn query_balance(
+            &self,
+            address: impl Into<String>,
+            denom: &str,
+        ) -> Result<Uint128, CwEnvError> {
+            Ok(self
+                .bank_querier()
+                .balance(address, Some(denom.to_string()))?
+                .first()
+                .map(|c| c.amount)
+                .unwrap_or_default())
+        }
+
+        /// Fetch all the balances of an address.
+        pub fn query_all_balances(
+            &self,
+            address: impl Into<String>,
+        ) -> Result<Vec<cosmwasm_std::Coin>, CwEnvError> {
+            self.bank_querier().balance(address, None)
+        }
     }
 
-    /// Fetch all the balances of an address.
-    pub fn query_all_balances(
-        &self,
-        address: impl Into<String>,
-    ) -> Result<Vec<cosmwasm_std::Coin>, CwEnvError> {
-        self.bank_querier().balance(address, None)
+    impl<S: StateInterface> MockBase<MockApiBech32, S> {
+        pub fn create_account(&self, account_name: impl Into<String>) -> Addr {
+            self.app.borrow().api().addr_make(&account_name.into())
+        }
     }
-}
 
-impl Mock<MockState> {
-    /// Create a mock environment with the default mock state.
-    pub fn new(sender: impl Into<String>) -> Self {
-        Mock::new_custom(sender, MockState::new())
+    impl Default for MockBase<MockApiBech32, MockState> {
+        fn default() -> Self {
+            MockBase::<MockApiBech32, MockState>::new_custom_bech32("mock", MockState::new())
+        }
     }
 
-    pub fn new_with_chain_id(sender: impl Into<String>, chain_id: &str) -> Self {
-        let chain = Mock::new_custom(sender, MockState::new());
-        chain
-            .app
-            .borrow_mut()
-            .update_block(|b| b.chain_id = chain_id.to_string());
+    impl<S: StateInterface> MockBase<MockApiBech32, S> {
+        /// Create a mock environment with a custom mock state.
+        /// The state is customizable by implementing the `StateInterface` trait on a custom struct and providing it on the custom constructor.
+        pub fn new_custom_bech32(prefix: &'static str, custom_state: S) -> Self {
+            let state = Rc::new(RefCell::new(custom_state));
+            let app = Rc::new(RefCell::new(
+                AppBuilder::new_custom()
+                    .with_api(MockApiBech32::new(prefix))
+                    .with_wasm(WasmKeeper::default().with_address_generator(MockAddressGenerator))
+                    .build(|_, _, _| {}),
+            ));
+
+            // We create an address internally
+            let sender = app.borrow().api().addr_make("sender");
+
+            Self { sender, state, app }
+        }
+    }
 
-        chain
+    impl<S: StateInterface> MockBech32<S> {
+        /// Set the bank balance of an address.
```

Comment:
Yes because we are chaging the way the address can be specified. For Bech32, we enforce passing Addr objects while for the normal api, any string will do. 
This simplifies the interface for users not using bech32